anydex/wallet/abstract_bitcoinlib_wallet.py

```diff
@@ -1,89 +1,134 @@
-          method in the __init__ method of the class (since we need access to the Tribler state directory) and
-          we can only import bitcoinlib *after* patching the bitcoinlib main file.
+    Used for throwing exceptions when a wallet is initialised with an invalid network
     """
-    TESTNET = False
+    def __init__(self, network):
+        super(UnsupportedNetwork, self).__init__(f'Network {network} is not supported.')
 
-    def __init__(self, wallet_dir):
-        super(BitcoinWallet, self).__init__()
 
-        bitcoinlib_main.initialize_lib(wallet_dir)
-        from bitcoinlib.wallets import wallet_exists, HDWallet
+class BitcoinlibWallet(Wallet):
+    """
+    Superclass used for the implementation of bitcoinlib wallets.
+    """
 
-        self.network = 'testnet' if self.TESTNET else 'bitcoin'
+    SUPPORTED_NETWORKS = ['bitcoin', 'litecoin', 'dash', 'litecoin_testnet', 'dash_testnet', 'testnet']
+
+    def __init__(self, wallet_dir, testnet, network, name, currency):
+        if network not in self.SUPPORTED_NETWORKS:
+            raise(UnsupportedNetwork(network))
+        super(BitcoinlibWallet, self).__init__()
+        self.testnet = testnet
+        self.network = network
+        self.currency = currency
+        self.name = name
         self.wallet_dir = wallet_dir
         self.min_confirmations = 0
         self.wallet = None
         self.unlocked = True
         self.db_path = os.path.join(wallet_dir, 'wallets.sqlite')
-        self.wallet_name = 'tribler_testnet' if self.TESTNET else 'tribler'
+        self.wallet_name = 'tribler testnet' if self.testnet else 'tribler_' + network
 
-        if wallet_exists(self.wallet_name, databasefile=self.db_path):
-            self.wallet = HDWallet(self.wallet_name, databasefile=self.db_path)
+        if wallet_exists(self.wallet_name, db_uri=self.db_path):
+            self.wallet_dir = HDWallet(self.wallet_name, db_uri=self.db_path)
             self.created = True
 
+        self.lib_init()
+
+    def cfg_init(self):
+        """
+        Adjusts the bitcoinlib configuration for the creation of a wallet
+        """
+        config = ConfigParser()
+
+        config['locations'] = {}
+        locations = config['locations']
+        locations['data_dir'] = self.wallet_dir.__str__()
+        # locations['database_dir'] = 'database'
+        # locations['default_databasefile'] = 'bitcoinlib.sqlite'
+        # locations['default_databasefile_cache'] = 'bitcoinlib_cache.sqlite'
+        locations['log_file'] = self.network + '_log.log'
+
+        config['common'] = {}
+        common = config['common']
+        # common['allow_database_threads'] = 'True'
+        # common['timeout_requests'] = '5'
+        # common['default_language'] = 'english'
+        common['default_network'] = self.network
+        # common['default_witness_type'] = ''
+        # common['service_caching_enabled'] = 'True'
+
+        config['logs'] = {}
+        # logs = config['logs']
```

Comment:
Are these comments still required to be present in the file?

Reply:
I thought it might be useful to leave them in case we decide to change something later on, so that we can just edit this file without looking at the _wrong_ sample config file provided.